expression/builtin_time.go

```diff
@@ -2023,35 +2023,36 @@ type makeDateFunctionClass struct {
 }
 
 func (c *makeDateFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinMakeDateSig{newBaseBuiltinFunc(args, ctx)}
+	tp := types.NewFieldType(mysql.TypeDate)
+	tp.Flen = 10
```

Comment:
@dawxy 
you may refer to [this](https://github.com/mysql/mysql-server/blob/5.7/sql/item_timefunc.h#L674)

[Clion](https://www.jetbrains.com/clion/) can be used to check MySQL source code.